### PR TITLE
Componentnize NewNoteButton

### PR DESCRIPTION
**Pull request by asmsuechan**
# context
Separation `NewNoteButton` from `TopBar`.

# before
image

image

## before(Trash)
8fb0f4979a82c703b4c74d14d61a1e6e

# after
I hope the layout isn't changed.
image

image

## after(Trash)
image

# for tests
* open modal
* create Markdown note
* create Snippet note
* show tooltip (and the content is correct on Mac or others)
* open modal (first-time Boostntoe launched)
* cannot be shown this in trash
* shown properly on dark theme and light theme
* shown properly even the width of the list is changed

# note
I just move the code, so I'll have to refactor them.

memo: delete DeleteArticleModal

Make sure whether all props are passed properly.

### CODE REVIEW

File: browser/main/NewNoteButton/index.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    }
+  }
+
+  resolveTargetFolder () {
+    let { data, params } = this.props
+    let storage = data.storageMap.get(params.storageKey)
+
+    // Find first storage
+    if (storage == null) {
+      for (let kv of data.storageMap) {
```
Comment (asmsuechan) [inline browser/main/NewNoteButton/index.js]: I want to seek better way

File: browser/main/NewNoteButton/index.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        storage = kv[1]
+        break
+      }
+    }
+    if (storage == null) window.alert('No storage to create a note')
```
Comment (asmsuechan) [inline browser/main/NewNoteButton/index.js]: I want to seek better way: window.alert

File: browser/main/NewNoteButton/index.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+    return {
+      storage,
+      folder
+    }
+  }
+
+  createNote (noteType) {
+    let { dispatch, location } = this.props
+    if (noteType !== 'MARKDOWN_NOTE' && noteType !== 'SNIPPET_NOTE') throw new Error('Invalid note type.')
```
Comment (asmsuechan) [inline browser/main/NewNoteButton/index.js]: Is this really needed?

File: browser/main/NewNoteButton/index.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+    let { storage, folder } = this.resolveTargetFolder()
+
+    let newNote = noteType === 'MARKDOWN_NOTE'
```
Comment (asmsuechan) [inline browser/main/NewNoteButton/index.js]: I want to seek better way

File: browser/main/NewNoteButton/index.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        })
+        hashHistory.push({
+          pathname: location.pathname,
+          query: {key: note.storage + '-' + note.key}
+        })
+        ee.emit('detail:focus')
+      })
+  }
+
+  setDefaultNote (defaultNote) {
```
Comment (asmsuechan) [inline browser/main/NewNoteButton/index.js]: Is this really used?

File: browser/main/NewNoteButton/NewNoteButton.styl [hunk 6]
```diff
@@ -0,0 +1,68 @@
+.root
+  position relative
+  background-color $ui-noteList-backgroundColor
+  height $topBar-height - 1
+  margin-left: auto;
+  width: 64px;
+  margin-right: -15px;
```
Comment (asmsuechan) [inline browser/main/NewNoteButton/NewNoteButton.styl]: I added these three lines for this change.

File: browser/main/NewNoteButton/index.js (hunk 5 above)
Comment (kazup01) [inline browser/main/NewNoteButton/index.js]: It was used in the old version, but not used now.

File: browser/main/NewNoteButton/NewNoteButton.styl (hunk 6 above)
Comment (kazup01) [inline browser/main/NewNoteButton/NewNoteButton.styl]: Thanks!

File: browser/main/NewNoteButton/index.js (hunk 5 above)
Comment (asmsuechan) [inline browser/main/NewNoteButton/index.js]: Understood.

File: browser/main/TopBar/index.js [hunk 7]
```diff
@@ -24,81 +21,19 @@ class TopBar extends React.Component {
...
-      dialog.showMessageBox(remote.getCurrentWindow(), {
-        type: 'warning',
-        message: 'Cannot create new note',
-        detail: 'You cannot create new note in trash box.',
-        buttons: ['OK']
-      })
-      return
-    }
-
-    switch (config.ui.defaultNote) {
```
Comment (asmsuechan) [inline browser/main/TopBar/index.js]: Deleted because `defaultNote` is not configurable in Preferences now.

Comment (kazup01) [inline browser/main/TopBar/index.js]: Thanks!

File: browser/main/TopBar/index.js [hunk 8]
```diff
@@ -147,60 +66,6 @@ class TopBar extends React.Component {
     }
   }
 
-  createNote (noteType) {
```
Comment (asmsuechan) [inline browser/main/TopBar/index.js]: Deleted accompanied with `handleNewPostButtonClick()`

File: browser/main/TopBar/index.js [hunk 9]
```diff
@@ -147,60 +66,6 @@ class TopBar extends React.Component {
...
-        })
-        hashHistory.push({
-          pathname: location.pathname,
-          query: {key: note.storage + '-' + note.key}
-        })
-        ee.emit('detail:focus')
-      })
-  }
-
-  setDefaultNote (defaultNote) {
```
Comment (asmsuechan) [inline browser/main/TopBar/index.js]: Deleted because it's no longer used.

File: browser/main/modals/DeleteArticleModal.js [hunk 10]
```diff
@@ -1,51 +0,0 @@
-import React, { PropTypes } from 'react'
```
Comment (asmsuechan) [inline browser/main/modals/DeleteArticleModal.js]: Deleted because this file is no longer used.

File: browser/main/TopBar/index.js (hunk 9 above)
Comment (kazup01) [inline browser/main/TopBar/index.js]: 👍

File: browser/main/NewNoteButton/index.js [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+    return {
+      storage,
+      folder
+    }
+  }
+
+  render () {
+    const { config, style, data, location } = this.props
+    if (location.pathname === '/trashed') {
```
Comment (kazup01) [inline browser/main/NewNoteButton/index.js]: Great🎉

**Comment by asmsuechan** [comment]
The layout is a bit strange.
image

File: browser/main/NewNoteButton/index.js (hunk 1 above)
Comment (asmsuechan) [inline browser/main/NewNoteButton/index.js]: I know this `if` condition is not good, but I don't know what exactly it is. So I choose a way not to change them.

**Pull request merged**